meta-mender-core/classes/mender-install-ubi.bbclass

```diff
@@ -0,0 +1,25 @@
+ARTIFACTIMG_FSTYPE ?= "ubifs"
+
+MENDER_STORAGE_DEVICE ?= "ubi0"
```

Comment:
Tricky indeed. I actually started out using `/dev` but gave up on it, and went with current implementation. Will try to explain in detail, and it will probably help me as well to decide what to do.

Ultimately what Mender client will use to update the volumes is `/dev/ubi0_0` and `/dev/ubi0_1`. And actually `/etc/fstab` can handle `/dev` as well, this was not the case last time I tested but I probably made a mistake first time around. So far so good :).

This problem with using `/dev` is then only that the kernel `root=` argument can not handle it. And this was the reason I dropped it initially as well.

To summarize, UBI volumes can be accessed/referenced in three different ways (this is Linux, u-boot only works with `ubi0:rootfsa` example but I all ready created a special variable for this and it is used only when we need to mount the file system to load kernel and dtb):

- /dev/ubi0_0
- ubi0_0
- ubi0:rootfsa

Above all mean the same, but none of these work everywhere :).

- /dev/ubi0_0
   - works in `/etc/fstab` (for data volume)
    - works for Mender client to update
    - does not work as `root=` argument

- ubi0_0
   - works in `/etc/fstab` (for data volume)
    - does not work for Mender client to use for update
    - works as `root=` argument.

- ubi0:rootfsa
   - works in `/etc/fstab` (for data volume)
   - does not work for Mender client to update
   - works as `root=` argument.

Above means that we need to do something special somewhere. Question is where does it make the most sense?